stan/math/rev/fun/gp_exp_quad_cov.hpp

```diff
@@ -208,82 +27,83 @@ class gp_exp_quad_cov_vari<T_x, double, T_l> : public vari {
  * @throw std::domain_error if sigma <= 0, l <= 0, or
  *   x is nan or infinite
  */
-template <typename T_x,
-          typename = require_arithmetic_t<typename scalar_type<T_x>::type>>
-inline Eigen::Matrix<var, -1, -1> gp_exp_quad_cov(const std::vector<T_x> &x,
-                                                  const var &sigma,
-                                                  const var &length_scale) {
-  check_positive("gp_exp_quad_cov", "sigma", sigma);
-  check_positive("gp_exp_quad_cov", "length_scale", length_scale);
-  size_t x_size = x.size();
-  for (size_t i = 0; i < x_size; ++i) {
+template <typename T_x, typename T_sigma, typename T_l,
+          require_stan_scalar_t<T_sigma>* = nullptr,
+          require_stan_scalar_t<T_l>* = nullptr,
```

Comment:
```suggestion
          require_all_stan_scalar_t<T_x, T_sigma, T_l>* = nullptr,
```

Reply:
T_x can be an Eigen type, so I'm just changing it for the other two.